tests/models/test_genericconfig.py

```diff
@@ -84,11 +82,19 @@ def test_attribute_set(generic_config: "CustomConfig") -> None:
     cast to this type. However, if it is a non-specified type (similar to a standard
     dict key/value-pair), then it should be fine to change the value type for that
     given key.
+    Note: Pydantic v2 will not coerce all types, e.g. any type to `str`.
+    One should instead use Annotated to create custom types.
+    See https://docs.pydantic.dev/dev-v2/usage/types/custom/ for more information.
 
     """
+    from pydantic import ValidationError
+
     generic_config.configuration["string"] = "bar"
-    generic_config.configuration["string"] = 3.14
-    assert generic_config.configuration["string"] == str(3.14)
+    # Will not work with Pydantic v2:
```

Comment:
this comment should be removed, or at least moved to the note at the beginning of the test